src/main/java/step3/Game.java

```diff
@@ -0,0 +1,40 @@
+package step3;
+
+import step3.dto.PositionResponse;
+import step3.dto.RaceRoundResult;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Game {
+    private final Cars cars;
+    private final RaceCount raceCount;
+
+    public Game(int carCount, int raceCount) {
+        this.cars = new Cars(carCount);
+        this.raceCount = new RaceCount(raceCount);
+    }
+
+    public Game(Cars cars, RaceCount raceCount) {
+        this.cars = cars;
+        this.raceCount = raceCount;
+    }
+
+    public List<RaceRoundResult> start() {
+        List<RaceRoundResult> responses = new ArrayList<>();
+
+        while (!raceCount.isDone()) {
+            List<Position> positions = raceCount.play(() -> cars.move());
+            responses.add(
+                    new RaceRoundResult(
+                            positions.stream()
+                                    .map(PositionResponse::new)
+                                    .collect(Collectors.toList())
+                    )
+            );
+        }
+
+        return responses;
+    }
```

Comment:
```suggestion
public List<RaceRoundResult> start() {
    List<RaceRoundResult> results = new ArrayList<>();
    
    while (!raceCount.isDone()) {
        List<Position> positions = raceCount.play(() -> cars.move());
        results.add(RaceRoundResult.from(positions));
    }
    
    return results;
}
```
- DTO 내부에서 매핑하는건 어떨까요? 
- 스트림 API와 비즈니스 로직이 섞이면 가독성이 많이 떨어집니다.